apps/prairielearn/src/pages/instructorAssessmentQuestions/components/EditQuestionModal.tsx

```diff
@@ -0,0 +1,491 @@
+  }, [question, alternativeGroup, isAlternative, assessmentType]);
+
+  // Determine which max property was originally set for auto grading (maxPoints vs maxAutoPoints)
+  // Prefer matching the points property (points → maxPoints, autoPoints → maxAutoPoints)
+  const originalMaxProperty = useMemo<'maxPoints' | 'maxAutoPoints'>(() => {
+    // Check own question first
+    if (question.maxAutoPoints !== undefined) return 'maxAutoPoints';
+    if (question.maxPoints !== undefined) return 'maxPoints';
+
+    // Check inherited from alternative group
+    if (isAlternative) {
+      if (alternativeGroup.maxAutoPoints !== undefined) return 'maxAutoPoints';
+      if (alternativeGroup.maxPoints !== undefined) return 'maxPoints';
+    }
+
+    // Default: match the points property
+    return originalPointsProperty === 'points' ? 'maxPoints' : 'maxAutoPoints';
+  }, [question, alternativeGroup, isAlternative, originalPointsProperty]);
+
+  const [localQuestion, setLocalQuestion] = useState<ZoneQuestionJson | QuestionAlternativeJson>(
+    () => structuredClone(question),
+  );
+
+  // Check if a value is inherited (only for alternatives)
+  const isInherited = (fieldName: keyof ZoneQuestionJson | keyof QuestionAlternativeJson) => {
+    if (!isAlternative) return false;
+    return (
+      (!(fieldName in localQuestion) ||
+        localQuestion[fieldName as keyof typeof localQuestion] === undefined) &&
+      fieldName in alternativeGroup &&
+      alternativeGroup[fieldName as keyof typeof alternativeGroup] !== undefined
+    );
+  };
+
+  // Check if points/autoPoints is inherited (special case since we track which one to use)
+  const isPointsInherited = () => {
+    if (!isAlternative) return false;
+    return (
+      localQuestion[originalPointsProperty] === undefined &&
+      alternativeGroup[originalPointsProperty] !== undefined
+    );
+  };
+
+  // Format auto points for display (handles both single numbers and arrays)
+  const formatAutoPoints = (value: number | number[] | undefined): string => {
+    if (value === undefined) return '';
+    if (Array.isArray(value)) return value.join(',');
+    return String(value);
+  };
+
+  // Parse auto points from string to array (only used on save, not on every keystroke)
+  const parseAutoPointsForSave = (value: string | number | number[]): number | number[] => {
+    // If it's already a number or array, return as-is
+    if (typeof value !== 'string') {
+      return value;
+    }
+
+    if (assessmentType === 'Exam') {
+      // For exams, parse string as array
+      const trimmed = value.trim();
+      if (trimmed === '') return [];
+      return trimmed
+        .split(',')
+        .map((v) => Number(v.trim()))
+        .filter((v) => !Number.isNaN(v));
+    }
+    // For homework, parse as single number
+    return Number(value);
+  };
+
+  // Get the auto points value to display (own or inherited)
+  // Uses the originalPointsProperty to determine which field to check
```

Comment:
You might consider using JSDoc comments for Intellisense on hover for this and similar functions.